borg/repository.py

```diff
@@ -712,8 +712,13 @@ def _read(self, fd, fmt, header, segment, offset, acceptable_tags):
             key = None
         else:
             raise TypeError("_read called with unsupported format")
-        if size > MAX_OBJECT_SIZE or size < fmt.size:
-            raise IntegrityError('Invalid segment entry size [segment {}, offset {}]'.format(
+        if size > MAX_OBJECT_SIZE:
+            # if you get this on an archive made with borg < 1.0.7 and many millions of files and
+            # you need to restore it, you can disable this check by using "if False:" above.
+            raise IntegrityError('Invalid segment entry size - too big [segment {}, offset {}]'.format(
+                segment, offset))
+        if size < fmt.size:
+            raise IntegrityError('Invalid segment entry size - too small [segment {}, offset {}]'.format(
```

Comment:
ah, i misunderstood you.
you did not mean the size of the header, but the total size.
ok, i'll add that.